ui/app/upload_image/controller.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const multer = require('multer')
+const httpRequest = require('request')
+const returnError = require('../errors/index')
+const logger = require('../logging/index')
+const uploadFile = multer({
+    destination: './',
+    limits: {
+        fileSize: 5000000 // limit file size to 5MB
+    }
+})
+
+// handles HTTP POST for the route /upload-image
+const handleUploadImageRoute = [
+    uploadFile.single('file'),
+    (request, response, next) => {
+        const token = request.session.jwt
+        const file = request.file
+        const userId = request.session.userId
+        handleUploadingImage(file, userId, token).then((image) => {
+            if (image['sloth_check'].contains_sloth) response.redirect('/image-library')
+            else {
+                response.status(200).render('index', {page: 'Slothbucket', contains_sloth: 'There\'s no sloth!'})
```

Comment:
You might want to call `contains_sloth` something else. The naming scheme implies it's gonna be a boolean variable. With a dynamically typed language you want to be careful of misnomers.

Reply:
contains_sloth holds a boolean value. It's true if the image contained a sloth, false otherwise. Therefore, I use it as a condition to see if I should redirect to the user's images (if true) or tell the user the image didn't have a sloth (if false). In this case, do you think renaming is still required?

Reply:
In the response, you're using it to hold a string value though. Am I misunderstanding something?

Reply:
"There's no sloth!"

Reply:
Gotcha. I thought you were talking about the if-statement. Will rename the variable sent to the template.